chain/network/src/peer/peer_actor.rs

```diff
@@ -844,21 +848,72 @@ impl StreamHandler<Result<Vec<u8>, ReasonForBan>> for PeerActor {
                         .map(|port| SocketAddr::new(self.peer_addr.ip(), port)),
                     account_id: None,
                 };
-                self.chain_info = handshake.sender_chain_info.clone();
-                self.peer_manager_wrapper_addr
-                    .send(ActixMessageWrapper::new_without_size(PeerToManagerMsg::RegisterPeer(RegisterPeer {
-                        actor: ctx.address(),
-                        peer_info: peer_info.clone(),
-                        peer_type: self.peer_type,
-                        chain_info: handshake.sender_chain_info.clone(),
+                self.connection_state = Some(Arc::new(ConnectedPeer {
```

Comment:
Nit: 

let's 

```
let connection_state = ...;
self.connection_state = Some(Arc::clone(&connection_state));
```

That'll avoid unwraps down the line.

Reply:
done